gcovr/tests/add_coverages/Makefile

```diff
@@ -0,0 +1,40 @@
+CFLAGS= -fprofile-arcs -ftest-coverage -fPIC
+
+all:
+	$(CXX) $(CFLAGS) -c foo.cpp -o foo.o
+	$(CXX) $(CFLAGS) -c bar.cpp -o bar.o
+	$(CXX) $(CFLAGS) -DFOO -c main.cpp -o main_foo.o
+	$(CXX) $(CFLAGS) main_foo.o foo.o -o testcase_foo
+	$(CXX) $(CFLAGS) -DBAR -c main.cpp -o main_bar.o
+	$(CXX) $(CFLAGS) main_bar.o bar.o -o testcase_bar
+
+run: txt xml html sonarqube json
+
+json_foo:
+	./testcase_foo
+	$(GCOVR) -d --json-pretty -o coverage_foo.json
+
+json_bar:
+	./testcase_bar
+	$(GCOVR) -d --json-pretty -o coverage_bar.json
+
+txt: json_foo json_bar
+	$(GCOVR) -a coverage_foo.json -a coverage_bar.json -o coverage.txt
+
+xml: json_foo json_bar
+	$(GCOVR) -a coverage_foo.json -a coverage_bar.json -x -o coverage.xml
+
+html: json_foo json_bar
+	$(GCOVR) -a coverage_foo.json -a coverage_bar.json --html-details -o coverage.html
+
+sonarqube: json_foo json_bar
+	$(GCOVR) -a coverage_foo.json -a coverage_bar.json --sonarqube sonarqube.xml
+
+json: json_foo json_bar
+	$(GCOVR) -a coverage_foo.json -a coverage_bar.json --json-pretty -o coverage.json
```

Comment:
The `json` make target will only be invoked if you add it to the list of targets in `gcovr/tests/test_gcovr.py`. The `run` target is not actually used by the test suite. Additionally, the test driver needs to be updated with `json=lambda x: x` in SCRUBBERS and `json='coverage*.json'` in OUTPUT_PATTERN.

Reply:
Ok, so I have to extend this line and add 'json':
`@pytest.mark.parametrize('format', ['txt', 'xml', 'html', 'sonarqube', 'json'])`
and next:

- extend all tests Makefiles to use json rule
- extend all reference directories with json coverage?

Reply:
I think the tests will fail if you don't update the other Makefiles to include a `json` target. But to keep this PR reasonably small, you don't have to create JSON reference files. The JSON target can be empty.

Reply:
I have added to Makefiles empty json target.